benchmarks/benchmarks/analysis/distances.py

```diff
@@ -11,120 +17,236 @@
+        self.ag2 = self.u.atoms[num_atoms:-num_atoms]
+        self.ag3 = self.u.atoms[-num_atoms:]
         self.allocated_array_2D = np.empty((num_atoms, num_atoms),
-                                            dtype=np.float64)
+                                           dtype=np.float64)
         self.array_shape_1D = int(num_atoms * (num_atoms - 1) / 2.)
         self.allocated_array_1D = np.empty(self.array_shape_1D,
                                            dtype=np.float64)
-        self.u = MDAnalysis.Universe(GRO)
+
+    def time_between(self, num_atoms):
+        """Benchmark determination of subgroup
+        of atomgroup that is within a specific
+        distance of two other atomgroups.
+        """
+        distances.between(group=self.ag3,
+                          A=self.ag1,
+                          B=self.ag2,
+                          distance=15.0)
+
+
+class DistancesBench(object):
+    """Benchmarks for MDAnalysis.analysis.distances
+    functions. Excluding contact matrices.
+    """
+    timeout = 180
+    params = ([10, 100, 1000, 10000], [None, 'orthogonal', 'triclinic'])
+    param_names = ['num_atoms', 'pbc_type']
+
+    def setup(self, num_atoms, pbc_type):
+
+        # Some functions are performance dependent on a realistic
+        # atom density. So we make a new universe with random atom
+        # positions, density derived from an example universe.
+
+        # Make fake universe with 2*num_atoms because some of our benchmarks
+        # require 2 atom groups.
+        universe_num_atoms = num_atoms*2
+        self.u = MDAnalysis.Universe.empty(universe_num_atoms, trajectory=True)
+        # calculating reasonable density for an atomistic system.
+        ex_universe = MDAnalysis.Universe(GRO)
+        density = ex_universe.coord.volume/ex_universe.atoms.n_atoms
+        desired_box_volume = universe_num_atoms*density
+        # making a cube with orthogonal basis vectors
+        cube_side_length = np.cbrt(desired_box_volume)
+        sides = [np.float32(cube_side_length)]*3
+        ortho_angles = [np.float(90)]*3
+        self.u.dimensions = np.array(sides + ortho_angles)
+
+        # filling the cube with random points.
+        np.random.seed(17809)
+        random_samples = np.random.random_sample((universe_num_atoms,
+                                                  3)).astype(np.float32)
+        self.u.atoms.positions = cube_side_length * random_samples
+
+        if pbc_type is None:
+            self.box_dims = None
+
+        elif pbc_type == 'orthogonal':
+            self.box_dims = self.u.dimensions
+
+        elif pbc_type == 'triclinic':
+            # making a triclinic box with the same volume as the cube
+            deg_alpha = np.float32(60)
+            deg_beta = np.float32(60)
+            deg_gamma = np.float32(60)
+            alpha = np.deg2rad(deg_alpha)
+            beta = np.deg2rad(deg_beta)
+            gamma = np.deg2rad(deg_gamma)
```

Comment:
I can't really think of a reason for all this repetition with `np.float32(60)` 3 times, etc. For example, `np.deg2rad()` takes a `dtype` argument anyway, and if there is only one value couldn't we just use a single `angle = np.deg2rad(60, dtype=np.float32)` variable or similar?

Reply:
Yeah this made more sense when I was debugging my code with different angle values. I'll change it.